src/storage/src/table/cell_based_table.rs

```diff
@@ -125,9 +125,47 @@ impl<S: StateStore> CellBasedTable<S> {
     }
 
     // cell-based interface
-    pub async fn get_row(&self, _pk: &Row, _epoch: u64) -> StorageResult<Option<Row>> {
-        // get row by state_store multi get
-        todo!()
+    pub async fn get_row(&self, pk: &Row, epoch: u64) -> StorageResult<Option<Row>> {
+        let pk_serializer = self.pk_serializer.as_ref().expect("pk_serializer is None");
+        let column_ids = generate_column_id(&self.column_descs);
+        // let mut row = Vec::with_capacity(column_ids.len());
+        let sentinel_key = [
+            &serialize_pk(pk, pk_serializer).map_err(err)?[..],
+            &serialize_column_id(&SENTINEL_CELL_ID).map_err(err)?,
+        ]
+        .concat();
+        let mut get_res = Vec::new();
+        let sentinel_cell = self.keyspace.get(&sentinel_key, epoch).await?;
+
+        if sentinel_cell.is_none() {
+            // if sentinel cell is none, this row doesn't exist
+            return Ok(None);
+        } else {
+            get_res.push((sentinel_key, sentinel_cell.unwrap()));
+        }
+        for column_id in column_ids {
+            let key = [
+                &serialize_pk(pk, pk_serializer).map_err(err)?[..],
+                &serialize_column_id(&column_id).map_err(err)?,
+            ]
+            .concat();
+            let state_store_get_res = self.keyspace.get(&key, epoch).await?;
+            if let Some(state_store_get_res) = state_store_get_res {
+                get_res.push((key, state_store_get_res));
+            }
+        }
+        let mut cell_based_row_deserializer =
+            CellBasedRowDeserializer::new(self.column_descs.clone());
```

Comment:
May optimize `CellBasedRowDeserializer` to take reference in the future, or refactor it to take `Arc<[ColumnDesc]>`.